Makefile

```diff
@@ -44,16 +44,25 @@ CFLAGS += $(DEBUGFLAGS) $(MOREFLAGS)
 FLAGS   = $(CFLAGS) $(CPPFLAGS)
 XXHSUM_VERSION = $(LIBVER)
 
+ifeq '$(findstring ;,$(PATH))' ';'
+    UNAME := Windows
+else
+    UNAME := $(shell uname 2>/dev/null || echo Unknown)
+    UNAME := $(patsubst CYGWIN%,Cygwin,$(UNAME))
```

Comment:
In a recent patch, I modified the OS detection below so that variable strings such as `CYGWIN%` can be used directly within the `filter()`

Reply:
Yup, but that puts all the match complexity down into every ifeq. I think it's nicer to simplify the values and thus make the follow-up conditionals easier. However, either way works fine.

Reply:
I agree for things which are used multiple times.
For example, defining `UNAME :=` once and reusing it everywhere is a neat idea.

But for the larger pattern test enabling the `make install` section, it's only invoked once.
So is it really better to define it elsewhere, rather than at the one place where it's used ?

Reply:
My latest merge just defines UNAME and fixes up the uninstall of xxh3.h.